packages/template-retail-react-app/app/hooks/use-datacloud.js

```diff
@@ -0,0 +1,481 @@
+     *
+     * This method records a `catalog` event type to track when a shopper
+     * views the details of a product (e.g. a Product Detail Page).
+     *
+     * @param {object} product - The product being viewed
+     * @param {object} args - Additional metadata for the event
+     */
+    async sendViewProduct(product, args) {
+        const baseEvent = this._constructBaseEvent(args)
+        const baseProduct = this._constructDatacloudProduct(product)
+
+        const identityProfile = this._concatenateEvents(
+            baseEvent,
+            this._generateEventDetails('identity', 'Profile'),
+            {
+                isAnonymous: args.isGuest,
+                firstName: args.firstName,
+                lastName: args.lastName
+            }
+        )
+
+        let contactPointEmail = null
+        if (args.email) {
+            contactPointEmail = this._concatenateEvents(
+                baseEvent,
+                this._generateEventDetails('contactPointEmail', 'Profile'),
+                {
+                    email: args.email
+                }
+            )
+        }
+
+        const catalog = this._concatenateEvents(
+            baseEvent,
+            this._generateEventDetails('catalog', 'Engagement'),
+            baseProduct,
+            {
+                type: 'Product',
+                webStoreId: 'pwa',
+                interactionName: 'catalog-object-view-start'
+            }
+        )
+
+        const interaction = {
+            events: [identityProfile, ...(contactPointEmail ? [contactPointEmail] : []), catalog]
+        }
+
+        try {
+            this.sdk.webEventsAppSourceIdPost(interaction)
+        } catch (err) {
+            logger.error('Error sending DataCloud event')
+        }
+    }
+
+    /**
+     * Sends a `catalog-object-impression` event to Data Cloud.
+     *
+     * This method records a `catalog` event type and represents a single
+     * page of product impressions (e.g. a Product List Page).
+     *
+     * One event is sent for each product on the page.
+     *
+     * @param {object} searchParams - The searchParams object
+     * @param {object} category - The category object
+     * @param {object} searchResults - The searchResults object
+     * @param {object} args - Additional metadata for the event
+     */
+    async sendViewCategory(searchParams, category, searchResults, args) {
+        const baseEvent = this._constructBaseEvent(args)
+
+        const products = searchResults?.hits?.map((product) =>
+            this._constructDatacloudProduct(product)
+        )
+
+        const catalogObjects = products.map((product) => {
+            return this._concatenateEvents(
+                baseEvent,
+                this._generateEventDetails('catalog', 'Engagement'),
+                this._constructBaseSearchResult(searchParams),
+                {
+                    id: product.id,
+                    type: 'Product',
+                    webStoreId: 'pwa',
```

Comment:
What does this id mean? Should it be siteId? 🤔

Reply:
I could be wrong, but in the schema doc, the The unique identifier of the catalog object is the categoryId, webstoreId is described as The web store associated with the event, it sound to me it is the siteId

Reply:
@alexvuong Yuanchen also informed me that because the schema is shared with Experience Cloud, some values used by their team does not have a specific usage for our case. So their PM asked them to use the `webStoreId` value to fill the `siteId` field as `siteId is a required field.

Reply:
If that is the case, should we use the value of `siteId` instead `pwa` string?